pkg/controller/uniregistrysyncer/uniregistrysyncer.go

```diff
@@ -0,0 +1,275 @@
+	controllerutil "github.com/openshift/ci-tools/pkg/controller/util"
+	"github.com/openshift/ci-tools/pkg/util/imagestreamtagmapper"
+	"github.com/openshift/ci-tools/pkg/util/imagestreamtagwrapper"
+)
+
+const ControllerName = "uni_registry_syncer"
+
+func AddToManager(appCIMgr manager.Manager,
+	apiCIMgr manager.Manager,
+	imageStreams sets.String,
+) error {
+	log := logrus.WithField("controller", ControllerName)
+	appCIClient := imagestreamtagwrapper.MustNew(appCIMgr.GetClient(), appCIMgr.GetCache())
+	apiCIClient := imagestreamtagwrapper.MustNew(apiCIMgr.GetClient(), apiCIMgr.GetCache())
+	r := &reconciler{
+		log:         log,
+		appCIClient: appCIClient,
+		apiCIClient: apiCIClient,
+	}
+	c, err := controller.New(ControllerName, appCIMgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: 1,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	if err := c.Watch(
+		source.NewKindWithCache(&imagev1.ImageStream{}, appCIMgr.GetCache()),
+		handlerFactory(imageStreamTagFilterFactory(log, imageStreams)),
+	); err != nil {
+		return fmt.Errorf("failed to create watch for ImageStreams: %w", err)
+	}
+
+	r.log.Info("Successfully added reconciler to manager")
+	return nil
+}
+
+type objectFilter func(types.NamespacedName) bool
+
+// handlerFactory produces a handler that:
+// * Watches ImageStreams because ImageStreamTags do not support the watch verb
+// * Extracts all ImageStramTags out of the Image
+// * Filters out the ones that are not in use
+// Note: We can not use a predicate because that is directly applied on the source and the source yields ImageStreams, not ImageStreamTags
+// * Creates a reconcile.Request per cluster and ImageStreamTag
+func handlerFactory(filter objectFilter) handler.EventHandler {
+	return imagestreamtagmapper.New(func(in reconcile.Request) []reconcile.Request {
+		if !filter(in.NamespacedName) {
+			return nil
+		}
+		return []reconcile.Request{in}
+	})
+}
+
+type reconciler struct {
+	log         *logrus.Entry
+	appCIClient ctrlruntimeclient.Client
+	apiCIClient ctrlruntimeclient.Client
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithField("request", req.String())
+	log.Info("Starting reconciliation")
+	err := r.reconcile(ctx, req, log)
+	// Ignore the logging for IsConflict errors because they are results of concurrent reconciling
+	if err != nil && !apierrors.IsConflict(err) && !apierrors.IsAlreadyExists(err) {
+		log.WithError(err).Error("Reconciliation failed")
+	} else {
+		log.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+const (
+	annotationDPTPRequester = "dptp.openshift.io/requester"
+)
+
+func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *logrus.Entry) error {
+	sourceImageStreamTag := &imagev1.ImageStreamTag{}
+	sourceClient := r.appCIClient
```

Comment:
why the `sourceClient` variable? The code below would be easier to understand if the `appCiClient` would be named like that instead of `sourceClient`

Reply:
Updated.
`app.ci` was too much alike `api.ci`.